File: app/src/main/java/knf/kuma/tv/cards/RelatedCardView.java
package knf.kuma.tv.cards;

import android.content.Context;
import android.net.Uri;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.commons.PicassoSingle;
import knf.kuma.database.CacheDB;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.tv.BindableCardView;

public class RelatedCardView extends BindableCardView<AnimeObject.WebInfo.AnimeRelated> {

    @BindView(R.id.img)
    ImageView imageView;
    @BindView(R.id.title)
    TextView title;
    @BindView(R.id.chapter)
    TextView type;

    public RelatedCardView(Context context) {
        super(context);
        ButterKnife.bind(this);
    }

    @Override
    public void bind(AnimeObject.WebInfo.AnimeRelated data) {
        AnimeObject object = CacheDB.INSTANCE.animeDAO().getByLink("%" + data.link);
        if (object != null)
            PicassoSingle.get(context).load(object.img).into(imageView);
        else
            PicassoSingle.get(context).load((Uri) null).into(imageView);
        title.setText(data.name);
        type.setText(data.relation);
    }

    @Override
    public ImageView getImageView() {
        return imageView;
    }

    @Override
    public int getLayoutResource() {
        return R.layout.item_tv_card_chapter;
    }
}

File: app/src/main/java/knf/kuma/tv/cards/RelatedCardView.kt
package knf.kuma.tv.cards

import android.content.Context
import android.widget.ImageView
import android.widget.TextView
import knf.kuma.R
import knf.kuma.commons.PatternUtil
import knf.kuma.commons.loadGlide
import knf.kuma.pojos.AnimeObject
import knf.kuma.pojos.av1.Relation
import knf.kuma.tv.BindableCardView
import org.jetbrains.anko.find

class RelatedCardView(context: Context) : BindableCardView<Relation>(context) {

    override val imageView: ImageView
        get() = find(R.id.img)
    override val layoutResource: Int
        get() = R.layout.item_tv_card_chapter

    override fun bind(data: Relation) {
        imageView.loadGlide(data.imageUrl)
        find<TextView>(R.id.title).text = data.name
        find<TextView>(R.id.chapter).text = getRelation(data.relation)
    }

    fun getRelation(id: Int): String {
        return when(id) {
            8 -> "Historia Principal"
            1 -> "Precuela"
            2 -> "Secuela"
            4 -> "Versión Alternativa"
            3 -> "Ambientación Alternativa"
            7 -> "Historia Completa"
            5 -> "Historia Paralela"
            6 -> "Resumen"
            else -> "Otro"
        }
    }
}
